### ISSUE TITLE
Use OIDC Discovery (.well-known/openid-configuration) to make this package usable with other providers

### ISSUE THREAD

**Issue by lauthieb** [auth0/auth0-spa-js#309]
### Describe the problem you'd like to have solved

[Decathlon](https://github.com/Decathlon) Front-End Community loves `auth0-spa-js` and all the documentation proposed by Auth0 to integrate it easily in our applications.  
However, we cannot use it because (e.g. for redirections), it is not designed to be used with a provider other than auth0.

After some research, I can confirm that redirections to `/authorize` and `/oauth/token` endpoints are hard coded in the codebase as you can see:
- [here](https://github.com/auth0/auth0-spa-js/blob/8ceb626cc01cf85b845a763bfffc789a2ecce250/src/Auth0Client.ts#L83) for authorize
- [here](https://github.com/auth0/auth0-spa-js/blob/8ceb626cc01cf85b845a763bfffc789a2ecce250/src/utils.ts#L191) for token

### Describe the ideal solution

It would be great to override the authorize & token endpoints by letting us enter our discovery `.well-known/openid-configuration` endpoint. (e.g. [our endpoint](https://idpdecathlon.oxylane.com/.well-known/openid-configuration))

### Alternatives and current work-arounds

For the moment, we don't imagine alternative. First we prefer to discuss here with you to find out if this is something we can consider.  
If you accept (mean we want to contribute to a more secure world together for any providers 😁), we may contribute through our [Decathlon Open Source organization](https://github.com/Decathlon).

**Thank you again for this library** ❤️

**Comment by stevehobbsdev** [comment]
Thanks for raising this. Generally, we would like to support more generic OIDC features such as using the discovery document. However, there may be some other things in the library that would need to be looked at in order to support this change. For example, how would internal workings of our silent authentication be affected (using an iframe, `prompt=none` and `response_mode=web_message`). Are those things that are generally supported by other IdPs? Is that something you can evaluate against your own provider?

There might be other things too, but having a PoC which can demonstrate integration with other providers would be a start.

**Comment by stevehobbsdev** [comment]
Closing this for now but if you would like to continue the discussion further, please feel free.

**Comment by lauthieb** [comment]
Sorry Steve for this delay of response but Christmas, recovery and health did not help me.

Anyway, thanks a lot for your detailed answer.
I have talked to developers in the Decathlon front-end community about this and unfortunately we will not have the bandwidth to process all of these subjects on your library.  

In addition, there is a huge naming job if it ever becomes multi-provider compatible because it would be weird to create a new Auth0Client with another provider.  

In short, I therefore effectively suggest that you close this issue and thank you very much for everything you do for the community.  

Have a nice day and we wish you a great new year 2020.
